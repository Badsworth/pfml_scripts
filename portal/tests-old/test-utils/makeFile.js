/**
 * Make a File instance
 * @see https://developer.mozilla.org/en-US/docs/Web/API/File
 * @param {object} attrs File attributes
 * @param {string} attrs.name The file's name
 * @param {string} attrs.type The file's MIME type
 * @returns {File}
 */
export const makeFile = (attrs = {}) => {
  const { name, type } = Object.assign(
    {
      name: "file.pdf",
      type: "application/pdf",
    },
    attrs
  );

  return new File([], name, { type });
};

/**
 * Create an HTML <input> for a test's event.target
 * @param {object} attrs - HTML attributes for the input
 * @returns {HTMLInputElement}
 */
export function createInputElement(attrs) {
  const input = document.createElement("input");

  Object.entries(attrs).forEach(([attributeName, value]) => {
    input.setAttribute(attributeName, value);
  });

  if (attrs.checked === false) input.checked = false;

  return input;
}
