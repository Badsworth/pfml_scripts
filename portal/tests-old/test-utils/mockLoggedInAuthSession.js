import { Auth } from "@aws-amplify/auth";

/**
 * Mock the Cognito authentication session so a test user appears logged in
 */
const mockLoggedInAuthSession = () => {
  const accessTokenJwt =
    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiQnVkIn0.YDRecdsqG_plEwM0H8rK7t2z0R3XRNESJB5ZXk-FRN8";

  jest.spyOn(Auth, "currentSession").mockImplementation(() =>
    Promise.resolve({
      accessToken: { jwtToken: accessTokenJwt },
    })
  );
};

export default mockLoggedInAuthSession;
