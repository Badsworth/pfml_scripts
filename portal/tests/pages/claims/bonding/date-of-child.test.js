import DateOfChild from "../../../../src/pages/claims/bonding/date-of-child";
import pick from "lodash/pick";
import { renderWithAppLogic } from "../../../test-utils";

jest.mock("../../../../src/hooks/useAppLogic");

describe("DateOfChild", () => {
  let appLogic, claim, wrapper;

  beforeEach(() => {
    ({ appLogic, claim, wrapper } = renderWithAppLogic(DateOfChild, {
      claimAttrs: {
        temp: {
          leave_details: {
            bonding: {
              date_of_child: "2020-08-01",
            },
          },
        },
      },
    }));
  });

  it("renders the page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when the form is successfully submitted", () => {
    it("calls claims.update", () => {
      wrapper.find("QuestionPage").simulate("save");

      expect(appLogic.claims.update).toHaveBeenCalledWith(
        expect.any(String),
        pick(claim, ["temp.leave_details.bonding.date_of_child"]),
        expect.any(Array)
      );
    });
  });
});
