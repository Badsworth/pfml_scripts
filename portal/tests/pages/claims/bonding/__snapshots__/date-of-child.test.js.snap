// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateOfChild renders the page 1`] = `
<QuestionPage
  onSave={[Function]}
  title="Leave details"
>
  <InputDate
    dayLabel="Day"
    example="For example: 04 / 28 / 1986"
    hint="If your child is not born or placed yet, enter the expected date. You can update this date later."
    label="What is the date of birth or placement date of your child?"
    monthLabel="Month"
    name="temp.leave_details.bonding.date_of_child"
    onChange={[Function]}
    value="2020-08-01"
    yearLabel="Year"
  />
</QuestionPage>
`;
